src/control/lib/hardware/pciutils/bindings.go

```diff
@@ -0,0 +1,132 @@
+//
+// (C) Copyright 2024 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package pciutils
+
+import (
+	"errors"
+	"os"
+	"unsafe"
+
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+)
+
+/*
+#cgo LDFLAGS: -lpci
+
+#include <stdlib.h>
+#include <pci/pci.h>
+*/
+import "C"
+
+type api struct {
+	access *C.struct_pci_access
+}
+
+func getAPI() (*api, error) {
+	api := &api{
+		access: C.pci_alloc(),
+	}
+
+	if api.access == nil {
+		return nil, errors.New("pci_alloc() failed")
+	}
+
+	return api, nil
+}
+
+func (api *api) Cleanup() {}
+
+func speedToFloat(speed uint16) float32 {
+	giga := 1000000000
+	tera := 1000 * giga
+
+	switch speed {
+	case 1:
+		return float32(giga) * 2.5
+	case 2:
+		return float32(tera) * 5
+	case 3:
+		return float32(tera) * 8
+	case 4:
+		return float32(tera) * 16
+	case 5:
+		return float32(tera) * 32
+	case 6:
+		return float32(tera) * 64
+	default:
+		return 0
+	}
+}
+
+func (api *api) PCIDeviceFromConfig(cfgBytes []byte, dev *hardware.PCIDevice) error {
```

Comment:
Looking at how we're using this now... I'd suggest either fleshing this out to fully-populate the incoming PCIDevice from the config, or, alternatively, renaming this to something like `PCICapsFromConfig`. I'm kind of inclined toward the latter, as I don't think we need a completely different way to populate a device. At least not for the moment.

Reply:
done